### PR TITLE
[RP-20] initial endpoint for CRUD class

### PR DESCRIPTION
**Pull request by c3bosskung**
https://nine0512.atlassian.net/browse/RP-19

please add those env variable for use mongo for test on your local

### CODE REVIEW

File: src/main/kotlin/com/rooplor/classcraftbackend/dtos/ClassList.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.rooplor.classcraftbackend.dtos
+
+import lombok.NoArgsConstructor
+
+@NoArgsConstructor
+data class ClassList(
```
Comment (Lostjerome) [inline src/main/kotlin/com/rooplor/classcraftbackend/dtos/ClassList.kt:6]: ClassListDTO ?

File: src/main/kotlin/com/rooplor/classcraftbackend/enums/Status.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.rooplor.classcraftbackend.enums
+
+enum class Status {
+    FILL_CASS_DETAIL,
+    RESERVE_VENUE,
+    CRAFT_YOUR_CONTENT,
```
Comment (Lostjerome) [inline src/main/kotlin/com/rooplor/classcraftbackend/enums/Status.kt:6]: CRAFT_CONTENT

File: src/main/kotlin/com/rooplor/classcraftbackend/enums/Type.kt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.rooplor.classcraftbackend.enums
+
+enum class Type {
```
Comment (Lostjerome) [inline src/main/kotlin/com/rooplor/classcraftbackend/enums/Type.kt:3]: ClassType

File: src/main/kotlin/com/rooplor/classcraftbackend/enums/Venue.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package com.rooplor.classcraftbackend.enums
+
+enum class Venue {
```
Comment (Lostjerome) [inline src/main/kotlin/com/rooplor/classcraftbackend/enums/Venue.kt:3]: Use `Venue` as new entity because of there're image, title, description, detail

**Review by Lostjerome — APPROVED** [review]
PR shouldn't has more than 5 file changes. Please separated them by it's topic.  Improve it for your good sake, this is serious :)

**Review by Lostjerome — CHANGES_REQUESTED** [review]
Fix from comments

File: src/test/kotlin/com/rooplor/classcraftbackend/services/ClassServiceTest.kt
Comment (Lostjerome) [inline src/test/kotlin/com/rooplor/classcraftbackend/services/ClassServiceTest.kt:1]: Would be better if we can provide realistic data

File: src/test/kotlin/com/rooplor/classcraftbackend/repositories/ClassRepositoryTest.kt
Comment (Lostjerome) [inline src/test/kotlin/com/rooplor/classcraftbackend/repositories/ClassRepositoryTest.kt:1]: Would be better if we can provide realistic data

File: src/main/kotlin/com/rooplor/classcraftbackend/utils/JsonValid.kt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper
+
+object JsonValid {
+    fun isValidJson(content: String): Boolean =
+        try {
+            ObjectMapper().readTree(content)
+            true
+        } catch (e: JsonProcessingException) {
+            false
+        }
```
Comment (Lostjerome) [inline src/main/kotlin/com/rooplor/classcraftbackend/utils/JsonValid.kt:13]: What it can do

Comment (c3bosskung) [reply src/main/kotlin/com/rooplor/classcraftbackend/utils/JsonValid.kt:13]: for validate json format when user create content

**Comment by Lostjerome** [comment]
คราวหน้าฝาก commit ให้ตรงกับ changes ข้างใน

**Comment by Lostjerome** [comment]
อันไหนแก้แล้ว ฝากกด resolve conversation ให้ด้วย

**Review by Lostjerome — APPROVED** [review]

**Pull request merged**